os/dnd.h

```diff
@@ -0,0 +1,108 @@
+// LAF OS Library
+// Copyright (C) 2024  Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifndef OS_DND_H_INCLUDED
+#define OS_DND_H_INCLUDED
+#pragma once
+
+#include "base/paths.h"
+#include "base/debug.h"
+#include "gfx/point.h"
+#include "os/surface.h"
+
+#include <memory>
+
+#pragma push_macro("None")
+#undef None // Undefine the X11 None macro
+
+namespace os {
+
+  class Window;
+
+  // Operations that can be supported by source and target windows in a drag
+  // and drop operation.
+  enum DropOperation {
```

Comment:
This should be an enum class:
```suggestion
  enum class DropOperation {
```

Reply:
Well...actually I've tried to use the enum class for this...but since this are flags and it is designed to be used with bitwise operators...I found it pretty inconvenient to have to implement all the bitwise operators to make it work as I wanted. So I opted to leave it as an enum. If you prefer the enum class along with the bitwise operators implementation, let me know and I can do it. Or maybe there is some other way to do this that I'm not aware of.

Reply:
Yes, `enum class` as flags are hard to use (a lot of `static_cast<int>` are required), but this must be done (in other case we have things like `os::Copy` or `os::None` available in the whole `os` namespace).

Reply:
Okay, so should we use the static_casts? Or should I implement the bitwise operators methods?

Reply:
Use static_cast when is needed.